### PR TITLE
Add JS SDK

### PR DESCRIPTION
**Pull request by wcgcyx**
This PR does three things:
1. Add a round trip to the integration test to test the 2-way bridge. 
2. Fix a small bug that is preventing the browser that runs JS SDK to pull proof from the message store.
3. Add an Initial version of the JS SDK that interacts with GPACT.

### CODE REVIEW

**Review by drinkcoffee — APPROVED** [review]

File: sdk/js/executor.js [hunk 1]
```diff
@@ -0,0 +1,300 @@
...
+                var eventID = getEventID(node.chainID, segmentEvent)
+                for (let j = 0; j < 60; j++) {
+                    var sig = await this.ms.getSignature(eventID)
+                    if (sig != null) {
+                        segmentEventSig = sig
+                        break
+                    }
+                    console.log("fail to get event sig, try again...")
+                    await new Promise(r => setTimeout(r, 2000))
+                }
```
Comment (ermyas) [inline sdk/js/executor.js:173]: If all retry attempts in contacting the message store fail here, then this block fails silently. Should the `err` variable be set to inform the caller? 

comment applies to `start()` and `root()` as well.

File: sdk/js/executor.js [hunk 2]
```diff
@@ -0,0 +1,300 @@
...
+        var temp = '0b'
+        for (let i = 0; i < 256; i++) {
+            temp += Math.round(Math.random())
+        }
+        var transID = BigInt(temp)
+
+        // Start event
+        console.log("start...")
+        var [startEvent, startEventSig, err] = await this.start(transID, root)
+        if (err != null) {
```
Comment (ermyas) [inline sdk/js/executor.js:36]: in the event contacting the message store fails inside `start()`, it appears that `err` will be null (see comment further down), even though the routine didn't fully succeed. Perhaps we can set `err` in those conditions inside the `start()` funciton, or check whether `startEventSign` is null here to detect whether the start routine was really successful?

File: sdk/js/executor.js [hunk 3]
```diff
@@ -0,0 +1,300 @@
...
+                    gas: 600000,
+                    gasPrice: gasPrice,
+                })
+                console.log("start succeed")
+                console.log(res)
+                startEvent = res.events.Start
+                // Try to get event sig
+                // Try 60 times, 2 seconds apart.
+                var eventID = getEventID(root.chainID, startEvent)
+                for (let j = 0; j < 60; j++) {
```
[the remark this answers is not in the document]
Comment (wcgcyx) [reply sdk/js/executor.js:92]: Sounds good

File: sdk/js/executor.js [hunk 4]
```diff
@@ -0,0 +1,300 @@
...
+            // This is an intermediate node.
+            // Append a 0 in the end of the call path.
+            callPath.push(0)
+            // Copy the call path
+            var childCallPath = callPath.slice()
+            // Need to collect segments event for all the child nodes.
+            for (let i = 0; i < node.children.length; i++) {
+                // Child call path last element starts with 1.
+                childCallPath[childCallPath.length - 1] = i + 1
+                var [childSegEvent, childSegSig, err2] = await this.segment(startChainID, startEvent, startEventSig, child, childCallPath, lockedSegments, lockedSegmentsSigs)
```
[the remark this answers is not in the document]
Comment (wcgcyx) [reply sdk/js/executor.js:132]: Yes, this is a bug

File: sdk/js/executor.js [hunk 5]
```diff
@@ -0,0 +1,300 @@
...
+
+export class Executor {
```
[the remark this answers is not in the document]
Comment (wcgcyx) [reply sdk/js/executor.js:13]: There is no unit testing at the moment. Should add unit testing for all the SDK components.

**Pull request merged**